libshortfin/CMakeLists.txt

```diff
@@ -43,10 +43,19 @@ set(SHORTFIN_IREE_SOURCE_DIR "" CACHE FILEPATH "Path to IREE source")
 # Options for building static or dynamic libraries.
 option(SHORTFIN_BUILD_STATIC "Builds static libraries" OFF)
 option(SHORTFIN_BUILD_DYNAMIC "Builds dynamic libraries" ON)
-cmake_dependent_option(SHORTFIN_LINK_DYNAMIC "Links internal binaries against static libshortfin.a" ON "SHORTFIN_BUILD_DYNAMIC" OFF)
 if(NOT SHORTFIN_BUILD_STATIC AND NOT SHORTFIN_BUILD_DYNAMIC)
   message(FATAL_ERROR "One of SHORTFIN_BUILD_STATIC or SHORTFIN_BUILD_DYNAMIC must be ON")
 endif()
+# Since dynamic build is the default option, if someone specifies they want
+# a static build, and both options are on, we should build a static build.
+if (SHORTFIN_BUILD_STATIC)
+  if (SHORTFIN_BUILD_DYNAMIC)
+    message(STATUS "Both SHORTFIN_BUILD_DYNAMIC and SHORTFIN_BUILD_STATIC specified, choosing static build")
```

Comment:
I tend to rather exit with a `FATAL_ERROR` and let the user fix the configuration instead of making assumptions here (similar to the above that one must be set).

Reply:
Looking closer to the CMake config, it seems desired that we can build both libraries. This wouldn't possible withbut is also prevented by this PR in general.